### PR TITLE
Add runtime containment cookbook for multi-agent workflows

### PR DESCRIPTION
**Pull request by amabito**
## Summary

Notebook showing how to add budget enforcement and circuit breaking to a
multi-agent loop built with the Agents SDK. Runs the same 3-agent workflow
with and without containment, then compares call count and execution time.

Tested ContainedRunner logic locally with 9 tests covering budget enforcement,
boundary conditions, circuit breaker open/reset, agent isolation, compound
state (budget + circuit simultaneous), and zero-budget edge case. All passing.

## Motivation

The agents_sdk cookbooks cover orchestration patterns (parallel, handoffs,
memory) but nothing covers runaway loops -- no budget cap, no circuit breaker,
no halt mechanism.

OpenAI's safety docs cover content guardrails. This is about the other half --
execution limits. Uses veronica-core as the containment layer, but any
budget/circuit-breaker library works.

---

## For new content

- [x] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [x] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [x] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [x] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [x] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [x] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [x] Correctness: The information I include is correct and all of my code executes successfully.
  - [x] Completeness: I have explained everything fully, including all necessary references and citations.

### CODE REVIEW

**Comment by amabito** [comment]
Addressed both findings:

**P1**: `raise` -> `return None` in the except block. The loop now continues and lets the circuit breaker accumulate failures to OPEN, which was the whole point of the demo.

**P2**: Added `report = ""` initialization before the loop in `run_uncontained`.

Force-pushed the fix.

File: examples/agents_sdk/runtime_containment_multi_agent.ipynb [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    "            self.call_count += 1\n",
+    "            # spend() records cost after the call completes -- the call\n",
+    "            # that crosses the limit still runs, but the next one is blocked\n",
+    "            self.budget.spend(0.05)  # estimate per call\n",
+    "            cb.record_success()\n",
+    "            return result\n",
+    "        except Exception:\n",
+    "            cb.record_failure()\n",
+    "            self.blocked_calls.append((name, f\"error (failures: {cb.failure_count})\"))\n",
+    "            raise"
```
[the remark this answers is not in the document]
Comment (amabito) [reply examples/agents_sdk/runtime_containment_multi_agent.ipynb]: Fixed in 58e5ef7. Changed raise to return None so the loop continues and the circuit breaker can accumulate.

File: examples/agents_sdk/runtime_containment_multi_agent.ipynb [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    "        result = await Runner.run(reviewer_agent, report)\n",
+    "        feedback = result.final_output\n",
+    "        stats[\"calls\"] += 1\n",
+    "\n",
+    "        print(f\"Iteration {i+1}: {len(report)} chars, {stats['calls']} total calls\")\n",
+    "        notes = f\"Previous report:\\n{report}\\n\\nReviewer feedback:\\n{feedback}\"\n",
+    "\n",
+    "    stats[\"elapsed\"] = time.time() - stats[\"start\"]\n",
+    "    stats[\"report_len\"] = len(report)\n",
+    "    return stats, report"
```
[the remark this answers is not in the document]
Comment (amabito) [reply examples/agents_sdk/runtime_containment_multi_agent.ipynb:147]: Fixed in 58e5ef7. Added report = "" before the loop.

**Comment by amabito** [comment]
@shikhar-cyber Your agentic governance cookbook covers content guardrails well. This notebook fills the other side -- execution limits (budget caps, circuit breaking) for agent loops. Meant to complement, not overlap.
